pkg/cache/cache_test.go

```diff
@@ -284,6 +294,80 @@ func CacheTest(createCacheFunc func(config *rest.Config, opts cache.Options) (ca
 					Expect(err).To(HaveOccurred())
 					Expect(errors.IsTimeout(err)).To(BeTrue())
 				})
+				type selectorsTestCase struct {
+					fieldSelectors map[string]string
+					labelSelectors map[string]string
+					expectedPods   []string
+				}
+				DescribeTable(" and cache with with selectors", func(tc selectorsTestCase) {
+					By("creating the cache")
+					builder := cache.BuilderWithOptions(
+						cache.Options{
+							SelectorsByObject: cache.SelectorsByObject{
+								&kcorev1.Pod{}: {
+									Label: labels.Set(tc.labelSelectors).AsSelector(),
+									Field: fields.Set(tc.fieldSelectors).AsSelector(),
+								},
+							},
+						},
+					)
+					informer, err := builder(cfg, cache.Options{})
+					Expect(err).NotTo(HaveOccurred())
+
+					By("running the cache and waiting for it to sync")
+					go func() {
+						defer GinkgoRecover()
+						Expect(informer.Start(informerCacheCtx)).To(Succeed())
+					}()
+					Expect(informer.WaitForCacheSync(informerCacheCtx)).NotTo(BeFalse())
+
+					obtainedPodList := kcorev1.PodList{}
```

Comment:
Can we also test the invariants of:
* List as unstructured.UnstructuredList
* List as MetadataList
?

Reply:
Done